### ISSUE TITLE
require_relative wrong in Jruby1.7 with symbolic links

### ISSUE THREAD

**Issue by savelandr** [jruby/jruby#3169]
#### Folder hierarchy:
```
$ tree
.
├── my_dir
│   ├── helper.rb
│   └── test.rb -> ../other_dir/test.rb
└── other_dir
    ├── helper.rb
    └── test.rb
```

#### Contents:
```
$ cat my_dir/helper.rb 
puts "helper in my_dir"

$ cat other_dir/helper.rb 
puts "helper in other_dir"

$ cat other_dir/test.rb 
require_relative 'helper'
```

### Jruby 1.7.14 test:

```
$ ruby -v
jruby 1.7.14 (1.9.3p392) 2014-08-27 d368971 on Java HotSpot(TM) 64-Bit Server VM 1.8.0_51-b16 +jit [linux-amd64]

$ ruby my_dir/test.rb 
helper in my_dir
```

### Jruby 9000 test:

```
$ ruby -v
jruby 9.0.0.0 (2.2.2) 2015-07-21 e10ec96 Java HotSpot(TM) 64-Bit Server VM 25.51-b03 on 1.8.0_51-b16 +jit [linux-amd64]

$ ruby my_dir/test.rb 
helper in other_dir
```

**Comment by enebo** [comment]
I typed this in weirdly and quickly so the output is a little different but I believe this is a bug in 1.7.  JRuby 9k matches MRI in both 2.2 and 1.9.3:

```text
system ~/work/snippets 1699% mri22 my_dir/test.rb 
OTHER HELP
system ~/work/snippets 1700% jruby my_dir/test.rb 
OTHER HELP
system ~/work/snippets 1701% mri19 my_dir/test.rb 
OTHER HELP
system ~/work/snippets 1702% ../jruby-1_7/bin/jruby my_dir/test.rb 
Heling in mydir
```

**Comment by savelandr** [comment]
Yup, I just installed 1.9.3 and 2.2.2 and they both behave the same as Jruby 9000.  Dang, I had gotten pretty used to this behavior...  Thanks!

**Comment by headius** [comment]
This is the same issue as #3092, fixed for 9k in 7e7292e632bd66a95fed78b37e73d6e63f90d631. I'll apply the require_relative fix for this bug as well as the File.realdirpath fix.

**Comment by headius** [comment]
This still needs a spec somewhere, but it has been fixed for 1.7.22.
